Administrator Guide/Thin-Arbiter-Volumes.md

```diff
@@ -0,0 +1,69 @@
+type of volume, quorum is taken into account at a brick level.
+If there is even one file/directory which is source on a particular
+brick, the other brick will not be allowed to become source for any
+other file/directory.
+
+<!-- TOC depthFrom:1 depthTo:6 withLinks:1 updateOnSave:1 orderedList:0 -->
+
+- [Why Thin Arbiter?](#why-thin-arbiter)
+- [How Thin Arbiter works](#how-thin-arbiter-works)
+
+<!-- /TOC -->
+
+# Thin Arbiter configuration
+
+The command to run thin-arbiter process on node:
+```
+#/usr/local/sbin/glusterfsd -N --volfile-id ta-vol -f /var/lib/glusterd/vols/thin-arbiter.vol --brick-port 24007 --xlator-option ta-vol-server.transport.socket.listen-port=24007
+```
+The syntax for creating the volume is:
+```
+#glustercli volume create <volname> --replica 2 Host1:Brick-1 Host2:Brick-2 --thin-arbiter host:/path-to-store-replica-id-file ...
+```
+_**Note:** Volumes using the arbiter feature can **only** be ```replica 3 arbiter 1```_
+
+For example:
+```
+glustercli volume create testvol --replica 2 server{1..2}:/bricks/brick-{1..2} --thin-arbiter server-3:/bricks/brick_ta --force
+volume create: testvol: success: please start the volume to access data
+```
+
+
+# Why Thin Arbiter?
+This is a solution for handling stretch cluster kind of workload
+but can be used for regular workloads as well in case the users are
+satisfied with this kind of quorum more than arbiter/3-way-replication.
+Cost will be lesser than arbiter. I/O will only go to the data bricks
+and goes to thin-arbiter only in the case of first failure until heal completes.
+
+
+# How Thin Arbiter works
+There is only one process running on thin arbiter node which manages all the replica pairs across all
+volumes. Each replica pair will use its respective replica-id file that is going to be created by
+glusterd for each of the replicas in all the replica volumes.
+
+1) Read Transactions:
+Reads are allowed when quorum is met. i.e.
+
+- When all bricks are up: Looking at the data bricks we can figure out which brick has good content
+  and serve that data to the client.
+- When one brick is up: Fail FOP with EIO.
+- Two bricks are up:
+  If 2 data bricks are up, perform lookup figure out good/bad bricks and serve content from the good brick.
+  Please note that in one generation one lookup is enough to figure out good/bad bricks. 1 data brick and
+  thin arbiter brick are up, do lookup on data brick, get info from thin arbiter and figure out if the data
+  brick has good/bad content. If the brick has good copy then serve data otherwise fail the FOP with EIO.
+
+2) Write transactions:
+  Thin arbiter doesn’t participate in I/O, transaction will choose to wind operations on thin-arbiter brick to
+  make sure the necessary metadata is kept up-to-date in case of failures. Operation failure will lead to
+  updating the replica-id file on thin-arbiter with source/sink information in the xattrs just how it happens in AFR.
```

Comment:
Okay. I think this document is targeted at developers, not users or admins, in which case it belongs in `docs` directory of glusterfs repo.

The glusterdocs repo is for users. It is best if you can simplify this and leave out the implementation details like winding FOPS, replica id etc

Reply:
I think it is good to have this much detail of fop transaction. At least admin should know all these things to use thin-arbiter. In case of failure this information will be helpful to understand the problem.

Reply:
I agree with Prashant and Sankarshan. This document needs to be restructured.

Reply:
@sankarshanmukhopadhyay @nbalacha I have raised a git hub issue for this  - #394 
@karthik-us will address the comments on this.